bin/demo.py

```diff
@@ -63,15 +66,47 @@ def stdchannel_redirected(stdchannel, dest_filename):
         if dest_file is not None:
             dest_file.close()
 
-def check_and_set_token():
+def needs_authentication():
+    token = dcos.config.get_config_val("core.dcos_acs_token")
+    if token is None:
+        # need to check if token is set because the CLI code will prompt for auth otherwise
+        return True
+    else:
+        try:
+            # make a request that requires authentication
+            shakedown.dcos_leader()
+            return False
+        except dcos.errors.DCOSException:
+            return True
+
+def authenticate_with_oauth(dcos_url, dcos_oauth_token):
+    dcos_oauth_token = arguments['--dcos-oauth-token']
+    url = dcos_url + 'acs/api/v1/auth/login'
+    creds = { 'token' : dcos_oauth_token }
```

Comment:
nit: is the spacing of the `:` correct here? Expected `'token': dcos_oauth_token`.